pkg/fleetautoscalers/fleetautoscalers.go

```diff
@@ -29,8 +36,64 @@ func computeDesiredFleetSize(fas *stablev1alpha1.FleetAutoscaler, f *stablev1alp
 	switch fas.Spec.Policy.Type {
 	case stablev1alpha1.BufferPolicyType:
 		return applyBufferPolicy(fas.Spec.Policy.Buffer, f)
+	case stablev1alpha1.WebhookPolicyType:
+		return applyWebhookPolicy(fas.Spec.Policy.Webhook, f)
 	}
 
+	return f.Status.Replicas, false, fmt.Errorf("Wrong Policy Type")
+}
+
+func applyWebhookPolicy(w *stablev1alpha1.WebhookPolicy, f *stablev1alpha1.Fleet) (int32, bool, error) {
+	faReq := v1alpha1.FleetAutoscaleReview{
+		Request: &v1alpha1.FleetAutoscaleRequest{
+			UID:       uuid.NewUUID(),
+			Name:      f.Name,
+			Namespace: f.Namespace,
+			Status:    f.Status,
+		},
+		Response: nil,
+	}
+	b, err := json.Marshal(faReq)
+	url := ""
+	if w.URL != "" {
+		url = w.URL
+	}
+	var faResp v1alpha1.FleetAutoscaleReview
+	if w.Service != nil {
+		if err != nil {
+			return f.Status.Replicas, false, err
+		}
+
+		if w.Service.Namespace == "" {
+			w.Service.Namespace = "default"
+		}
+		url = fmt.Sprintf("http://%s.%s.svc:8000/%s", w.Service.Name, w.Service.Namespace, w.Service.Path)
+	}
+	if url == "" {
+		return f.Status.Replicas, false, fmt.Errorf("URL was not provided")
+	}
+	res, err := http.Post(
+		url,
+		"application/json",
+		strings.NewReader(string(b)),
+	)
+	if res.StatusCode != http.StatusOK {
+		return f.Status.Replicas, false, fmt.Errorf("Bad Status code from the server")
+	}
+	if err != nil {
+		return f.Status.Replicas, false, err
+	}
+	result, err := ioutil.ReadAll(res.Body)
```

Comment:
You're leaking a connection see : 

- https://golang.org/pkg/net/http/
- https://stackoverflow.com/questions/33238518/what-could-happen-if-i-dont-close-response-body-in-golang

```suggestion
	defer res.Body.Close()
	result, err := ioutil.ReadAll(res.Body)
```

Reply:
Thanks, overseen this